service-broker/src/main/java/io/enmasse/osb/api/bind/OSBBindingService.java

```diff
@@ -196,12 +196,44 @@ private String generatePassword() {
     public Response unbindServiceInstance(@Context SecurityContext securityContext, @PathParam("instanceId") String instanceId, @PathParam("bindingId") String bindingId) {
         log.info("Received unbind request for instance {}, binding {}", instanceId, bindingId);
         verifyAuthorized(securityContext, ResourceVerb.get);
-/*
-        AddressSpace addressSpace = findAddressSpaceByAddressUuid(instanceId)
-                .orElseThrow(() -> OSBExceptions.notFoundException("Service instance " + instanceId + " does not exist"));
-*/
 
-        return Response.ok(new EmptyResponse()).build();
+        AddressSpace addressSpace = findAddressSpaceByInstanceId(instanceId)
+            .orElseThrow(() -> Exceptions.notFoundException("Service instance " + instanceId + " does not exist"));
+
+        try {
+            String username = "user-" + bindingId;
+
+            if(deleteUser(addressSpace, username)) {
+                return Response.ok(new EmptyResponse()).build();
+            } else {
+                return Response.status(Response.Status.GONE).build();
+            }
+        } catch (Exception e) {
+            throw new InternalServerErrorException("Exception interacting with auth service", e);
+        }
+
+    }
+
+    private boolean deleteUser(AddressSpace addressSpace, String username) throws Exception {
+            UserSpec.Builder specBuilder = new UserSpec.Builder();
+            specBuilder.setUsername(username);
+            User user = new User.Builder()
+                .setMetadata(new UserMetadata.Builder()
+                                 .setNamespace(addressSpace.getNamespace())
+                                 .setName(addressSpace.getName() + "." + username)
+                                 .build())
+                .setSpec(specBuilder.build())
+                .build();
+
+            String realmName = addressSpace.getAnnotation(AnnotationKeys.REALM_NAME);
+            if (userApi.getUserWithName(realmName, username).isPresent()) {
```

Comment:
Why not  use the user returned here in the call to delete instead of constructing one?

Reply:
because I was doing a quick C&P without actually looking at the code :-)

I did notice this could be simplified after I pushed it